background/messages/utxo-request.ts

```diff
@@ -0,0 +1,243 @@
+import type { MessagesMetadata, PlasmoMessaging } from "@plasmohq/messaging";
+import { JsonRpcProvider } from "ethers";
+import { ChainKey, chains, RequestMethod, rpcUrl } from "~utils/constants";
+import type { Messaging, TransactionProps } from "~utils/interfaces";
+import { v4 as uuidv4 } from "uuid";
+import {
+  getStoredTransactions,
+  getStoredVaults,
+  setStoredRequest,
+  setStoredTransactions,
+  setStoredVaults,
+} from "~utils/storage";
+import api from "~utils/api";
+
+let rpcProvider: JsonRpcProvider;
+
+const initializeProvider = (chainKey: string) => {
+  const rpc = rpcUrl[chainKey];
+  rpcProvider = new JsonRpcProvider(rpc);
+};
+
+const getAccounts = (
+  chain: ChainKey,
+  sender: string
+): Promise<{ accounts: string[] }> => {
+  return new Promise((resolve) => {
+    setStoredRequest({
+      chain,
+      sender,
+    }).then(() => {
+      let createdWindowId: number;
+
+      chrome.windows.getCurrent({ populate: true }, (currentWindow) => {
+        const height = 639;
```

Comment:
@antoni0dev 
Thanks for the comment. Im creating another PR for doing a lot of cleanup and removing duplicate code. Lets leave if for the other PR because there are lots of cleanup and my PR is almost ready. I will share it with you after this PR get merged